File: 274.py
# LeetCode 274 H-index
# Reference: https://leetcode.com/problems/h-index/solutions/4928640/python-2-approaches-sorting-counting-summary-with-comparison/?envType=study-plan-v2&envId=top-interview-150

class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        citations.sort()

        for i, v in enumerate(citations):
            if n - 1 <= v:
                return n - i
        return 0

File: 274.java
// LeetCode 274 H-index
// Reference: https://leetcode.com/problems/h-index/solutions/70768/java-bucket-sort-o-n-solution-with-detail-explanation/?envType=study-plan-v2&envId=top-interview-150

class Solution {
    public int hIndex(int[] citations) {

        int n = citations.length;
        int[] buckets = new int[n + 1];

        for(int c : citations){

            if (c >= n){
                buckets[n]++;

            } else {
                buckets[c]++;
            }
        }   
        int count = 0;
        for (int i = n; i >= 0; i--){
            count += buckets[i];
            if(count >= i){
                return i;
            }
        }
        return 0;
    }
}
